### ISSUE TITLE
flatpak-pip-generator fail on jupyter

### ISSUE THREAD

**Issue by pums974** [flatpak/flatpak-builder-tools#34]
```
$> python3 flatpak-pip-generator jupyter
Collecting jupyter
  Using cached https://files.pythonhosted.org/packages/c9/a9/371d0b8fe37dd231cf4b2cff0a9f0f25e98f3a73c3771742444be27f2944/jupyter-1.0.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/jupyter-1.0.0.tar.gz
Collecting notebook (from jupyter)
  Using cached https://files.pythonhosted.org/packages/cb/f9/e7b0df5cc36286ccafb7a70e23a9dcf9a1c2ee961dcd18509db3757639bf/notebook-5.6.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/notebook-5.6.0.tar.gz
Collecting qtconsole (from jupyter)
  Using cached https://files.pythonhosted.org/packages/e4/6d/5af06751ece05ff0892dd02ce043062bc2d8f426f91894876f669a70ef1b/qtconsole-4.4.1.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/qtconsole-4.4.1.tar.gz
Collecting jupyter-console (from jupyter)
  Using cached https://files.pythonhosted.org/packages/cd/bb/fa8901715c009a00ea4c42897da8585189d4781c64f4617a8bd65ca01a3a/jupyter_console-5.2.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/jupyter_console-5.2.0.tar.gz
Collecting nbconvert (from jupyter)
  Using cached https://files.pythonhosted.org/packages/88/68/0417164f344ad702245424d8f998b5c99f1c6147ad1ad645a465ae932134/nbconvert-5.4.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/nbconvert-5.4.0.tar.gz
Collecting ipykernel (from jupyter)
  Using cached https://files.pythonhosted.org/packages/83/2d/1e9860d470725ef22a2c06c15d082ca06d425e91eba1a1292bb0165af128/ipykernel-4.9.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/ipykernel-4.9.0.tar.gz
Collecting ipywidgets (from jupyter)
  Using cached https://files.pythonhosted.org/packages/88/8d/c5d3ac1bdcd74c7327c733c9d02354ff12d7d72744f262ed1e26560962b2/ipywidgets-7.4.2.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/ipywidgets-7.4.2.tar.gz
Collecting jinja2 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/56/e6/332789f295cf22308386cf5bbd1f4e00ed11484299c5d7383378cf48ba47/Jinja2-2.10.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/Jinja2-2.10.tar.gz
Collecting tornado>=4 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/e6/78/6e7b5af12c12bdf38ca9bfe863fcaf53dc10430a312d0324e76c1e5ca426/tornado-5.1.1.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/tornado-5.1.1.tar.gz
Collecting pyzmq>=17 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/b9/6a/bc9277b78f5c3236e36b8c16f4d2701a7fd4fa2eb697159d3e0a3a991573/pyzmq-17.1.2.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/pyzmq-17.1.2.tar.gz
Collecting ipython_genutils (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/e8/69/fbeffffc05236398ebfcfb512b6d2511c622871dca1746361006da310399/ipython_genutils-0.2.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/ipython_genutils-0.2.0.tar.gz
Collecting traitlets>=4.2.1 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/a5/98/7f5ef2fe9e9e071813aaf9cb91d1a732e0a68b6c44a32b38cb8e14c3f069/traitlets-4.3.2.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/traitlets-4.3.2.tar.gz
Collecting jupyter_core>=4.4.0 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/b6/2d/2804f4de3a95583f65e5dcb4d7c8c7183124882323758996e867f47e72af/jupyter_core-4.4.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/jupyter_core-4.4.0.tar.gz
Collecting jupyter_client>=5.2.0 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/4c/df/1e8df7f4de63cc667a7a9aa234539c0419513bf94ac57d36d73b3b434786/jupyter_client-5.2.3.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/jupyter_client-5.2.3.tar.gz
Collecting nbformat (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/6e/0e/160754f7ae3e984863f585a3743b0ed1702043a81245907c8fae2d537155/nbformat-4.4.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/nbformat-4.4.0.tar.gz
Collecting Send2Trash (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/13/2e/ea40de0304bb1dc4eb309de90aeec39871b9b7c4bd30f1a3cdcb3496f5c0/Send2Trash-1.5.0.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/Send2Trash-1.5.0.tar.gz
Collecting terminado>=0.8.1 (from notebook->jupyter)
  Using cached https://files.pythonhosted.org/packages/67/84/ce0ebd0f60e1cbe040f8e065eef7063855d59d9cf5e6438b3f8439fc7e15/terminado-0.8.1.tar.gz
  Saved /tmp/pip-generator-python3-jupyter-oe44xfx0/terminado-0.8.1.tar.gz
  Missing build requirements in pyproject.toml for terminado>=0.8.1 from https://files.pythonhosted.org/packages/67/84/ce0ebd0f60e1cbe040f8e065eef7063855d59d9cf5e6438b3f8439fc7e15/terminado-0.8.1.tar.gz#sha256=55abf9ade563b8f9be1f34e4233c7b7bde726059947a593322e8a553cc4c067a (from notebook->jupyter).
  This version of pip does not implement PEP 517 so it cannot build a wheel without 'setuptools' and 'wheel'.
  Installing build dependencies ... error
  Complete output from command /usr/bin/python -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-o62aba1e --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- flit:
  Collecting flit
    Using cached https://files.pythonhosted.org/packages/72/9d/b8f00445bb458a15c670308cf30e08900dc749690f299b7539ab70273ac2/flit-1.1.tar.gz
    Missing build requirements in pyproject.toml for flit from https://files.pythonhosted.org/packages/72/9d/b8f00445bb458a15c670308cf30e08900dc749690f299b7539ab70273ac2/flit-1.1.tar.gz#sha256=da823d4acae9bda42dcc0c7ab1d9be475a8a47aae5fd6dde63841d9f430ccb2f.
    This version of pip does not implement PEP 517 so it cannot build a wheel without 'setuptools' and 'wheel'.
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
      Complete output from command python setup.py egg_info:
      error in flit setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
  
      ----------------------------------------
  Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-lrou1uq5/flit/
  
  ----------------------------------------
Command "/usr/bin/python -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-o62aba1e --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- flit" failed with error code 1 in None
Traceback (most recent call last):
  File "flatpak-pip-generator", line 74, in <module>
    ] + opts.packages, check=True)
  File "/usr/lib/python3.7/subprocess.py", line 468, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['pip3', 'download', '--dest', '/tmp/pip-generator-python3-jupyter-oe44xfx0', '--no-binary', ':all:', 'jupyter']' returned non-zero exit status 1.
```

Using 
```
pip 18.0 from /usr/lib/python3.7/site-packages/pip (python 3.7)
```

**Comment by ghisvail** [comment]
```
  Collecting flit
    Using cached https://files.pythonhosted.org/packages/72/9d/b8f00445bb458a15c670308cf30e08900dc749690f299b7539ab70273ac2/flit-1.1.tar.gz
    Missing build requirements in pyproject.toml for flit from https://files.pythonhosted.org/packages/72/9d/b8f00445bb458a15c670308cf30e08900dc749690f299b7539ab70273ac2/flit-1.1.tar.gz#sha256=da823d4acae9bda42dcc0c7ab1d9be475a8a47aae5fd6dde63841d9f430ccb2f.
    This version of pip does not implement PEP 517 so it cannot build a wheel without 'setuptools' and 'wheel'.
```

Have you got `python3-setuptools` and `python3-wheel` installed on your machine?

**Comment by pums974** [comment]
Yes I have
and flatpak-pip-generator works well for many other packages on my computer,
so I don't think I'm missing something

**Comment by danyeaw** [comment]
Fixed by #63.
